### ISSUE TITLE
The product of the ParallelStrategyConfig is allowed to be less then the number of GPUs

### ISSUE THREAD

**Issue by imyzx2017** [flexflow/FlexFlow#225]
Hi, I have read your source code about strategy_search_task. When I read the code of func `get_random_parallel_config` in `linear.cu`, I have a question to ask:
```
for (int i = 1; i <= ff.config.workersPerNode; i++)
    if (channel % i == 0)
      for (int j = 1; i * j <= total_devices; j++)
        if (batch % j == 0) {
          batch_candidates.push_back(j);
          channel_candidates.push_back(i);
```
1. when run the func `get_random_parallel_config` of current ops(Linear), the product of the new rewrite ParallelStrategyConfig is allowed to be less then the number of GPUs  (`i*j <= total_devices`). Is this reasonable, will the data and model parameters assigned to some cards be the same? How do you solve this situation, thanks for your reply!

**Comment by jiazhihao** [comment]
You are right that FlexFlow allows using a subset of GPUs if that's beneficial (e.g., communication cost outweighs performance gains). You can image data and model parallelism as two dimensions FlexFlow considers for parallelizing Linear operators. In this case, the product of the degree of data and model parallelism is the overall number of GPUs used for training a Linear op.
